apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -401,24 +413,39 @@ void AprilTagDetector::imageCb(const sensor_msgs::PointCloud2ConstPtr& cloud,
           auto description = begin->second->descriptionsQueue_.front();
           tf::poseStampedMsgToTF(tag_pose, tag_transform);
           tf_pub_.sendTransform(tf::StampedTransform(tag_transform, tag_transform.stamp_, tag_transform.frame_id_, description.frame_name()));
-          tag_detection_array.detections.push_back(tag_detection);
+          valid_pose_tag_detection_array.detections.push_back(tag_detection);
         }
 
         begin->second->posesQueue_.pop_front();
         begin->second->descriptionsQueue_.pop_front();
       }
     }
 
+    for (auto && tag_detection : all_tracked_april_tags_) {
+      auto tag = tag_detection.second->posesQueue_[0];
+      all_tag_detection_array.detections.push_back(tag);
```

Comment:
Can't we just push tags directly into all_tag_detection_array and empty it after every publish? Is there a benefit to going through the queue first? Seems like tags will stick around in the queue until they age out, but will be overwritten if the same tag appears multiple times. Is this the desired behavior?

Reply:
This expected behavior. We are not guaranteed to see all the tags all the time so we keep them around in order to avoid spurious appear/disappear of detections.

Reply:
But doesn't that mean that old detections will keep getting republished at the same robot-relative pose even if the robot has since moved? This means that someone swinging the robot past the april tag on the way to putting it in the fixture could still get a passed test even though the tag is out of view when they start the test.